HealthcareSystem/Backend/Service/DrugConsumptionService/DrugConsumptionService.cs

```diff
@@ -0,0 +1,47 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using Backend.Model.Pharmacies;
+using Backend.Repository.DrugConsumptionRepository;
+
+namespace Backend.Service.DrugConsumptionService
+{
+    public class DrugConsumptionService : IDrugConsumptionService
+    {
+        private readonly IDrugConsumptionRepository _drugConsumptionRepository;
+
+        public DrugConsumptionService(IDrugConsumptionRepository drugConsumptionRepository)
+        {
+            _drugConsumptionRepository = drugConsumptionRepository;
+        }
+        public List<DrugConsumption> GetAllDrugConsumptions()
+        {
+            return _drugConsumptionRepository.GetAllDrugConsumptions().ToList();
+        }
+
+        public DrugConsumption GetDrugConsumptionById(int id)
+        {
+            return _drugConsumptionRepository.GetDrugConsumptionById(id);
+        }
+
+        public void CreateDrugConsumption(DrugConsumption dc)
+        {
+            _drugConsumptionRepository.CreateDrugConsumption(dc);
```

Comment:
I don't think this should just be created on it's own like this. The Drug class has the quantity field in it - so really, the way I understand this use case, the saved DrugConsumption instances should be a historical record of how that field changed. So I feel like the more correct business logic would be something like this - you have a service that deals with the hospital's drug storage in general. It has methods to reduce and increase the current quantity of a Drug, which can also create a DrugConsumption as a side effect. And then it would also make sense to me to put getters for things like current quantity of a drug and DrugConsumptions and DrugConsumptionRecords there.
I don't think this use case required you to actually implement this business logic though, you really just need the getters to make reports? So I probably would've preferred just adding the methods you need at this point.

Reply:
I agree that we don't need methods such as this right now. Having DrugConsumptions table be updated from other services such as general Drug service or some other services that deal with our stock seems like a much better idea. For now I have removed these unnecessary CRUD operations as they are not needed for this feature. At some point later if someone is working on a feature that involves changing drug quantity in the storage, they could also just tie in DrugConsumptions.

Reply:
Fixed in 6498babbb3b08e9824f980dab9090775e0555339.